cmd/machine-controller/main.go

```diff
@@ -198,14 +196,6 @@ func main() {
 	if err := clusterv1alpha1.AddToScheme(scheme.Scheme); err != nil {
 		klog.Fatalf("failed to add clusterv1alpha1 api to scheme: %v", err)
 	}
-	// Check if the hyperkube image has a tag set
-	hyperkubeImageRef, err := reference.Parse(nodeHyperkubeImage)
-	if err != nil {
-		klog.Fatalf("failed to parse --node-hyperkube-image %s: %v", nodeHyperkubeImage, err)
-	}
-	if _, ok := hyperkubeImageRef.(reference.NamedTagged); ok {
-		klog.Fatalf("--node-hyperkube-image must not contain a tag. The tag will be dynamically set for each Machine.")
-	}
```

Comment:
We should show a warning if the flag is provided that setting the hyperkube image is deprecated.

Reply:
👍